packages/common/src/util.ts

```diff
@@ -19,10 +19,15 @@ export function cloneObject<T>(obj: T): T {
     clone = obj.map(cloneObject);
   } else if (typeof obj === "object") {
     for (const i in obj) {
-      if (obj[i] != null && typeof obj[i] === "object") {
-        clone[i] = cloneObject(obj[i]);
-      } else {
-        clone[i] = obj[i];
+      /* istanbul ignore next no need to deal w/ other side of hasOwnProperty() */
+      if (obj.hasOwnProperty(i)) {
+        const value = obj[i];
+        const isDate = value instanceof Date;
+        if (value != null && typeof value === "object" && !isDate) {
+          clone[i] = cloneObject(value);
+        } else {
+          clone[i] = value;
```

Comment:
Since Date is an object I'd expect the clone and original to share a ref to the same Data instance... I think we'd want to convert it to a timestamp then create a new date instance from that... no?

Reply:
good catch @dbouwman - i updated the code as you suggest and updated the test to ensure it's not a reference to the the original